### ISSUE TITLE
OidcClient.LoginAsync() Method not found Error

### ISSUE THREAD

**Issue by b-esf** [DuendeArchive/IdentityModel.OidcClient#58]
Hello,

I'm trying to implement IdentityModel.OidcClient for our WinForms solution in order to authenticate against a local IdentityServer. I followed the provided WinForm sample (https://github.com/IdentityModel/IdentityModel.OidcClient.Samples/tree/master/WinFormsWebView) and included the WebView .cs files in my project. This is what my code looks like:

```
var options = new IdentityModel.OidcClient.OidcClientOptions
                {
                    Authority = _authority,
                    ClientId = _clientId,
                    Scope = _scope,
                    RedirectUri = _redirectUri,

                    Browser = new WinFormsEmbeddedBrowser()
                };

IdentityModel.OidcClient.OidcClient _oidcClient = new IdentityModel.OidcClient.OidcClient(options);
var result = await _oidcClient.LoginAsync(IdentityModel.OidcClient.Browser.DisplayMode.Visible);
```

When I try to go past the last line with LoginAsync(), I get the following error:

`Method not found: 'Void IdentityModel.Client.DiscoveryClient..ctor(System.String, System.Net.Http.HttpMessageHandler)'.`

I have <AutoGenerateBindingRedirects> enabled in my .csproj file, which generated the following redirects:
```
<dependentAssembly>
        <assemblyIdentity name="Microsoft.Ink" publicKeyToken="31bf3856ad364e35" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-6.1.0.0" newVersion="6.1.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-11.0.0.0" newVersion="11.0.0.0" />
      </dependentAssembly>
      <dependentAssembly>
        <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.1.1.2" newVersion="4.1.1.2" />        
</dependentAssembly>
```
I tried to recreate the issue in a new console project with just the IdentityModel.OidcClient nuget package, WebView files for the browser, and the same code snippet as above. When I run it, the program just exits as soon as it hits the LoginAsync() part of the code without warnings or any exceptions.

I've been banging my head against the wall on this for a couple of days, so any help would be greatly appreciated! Thanks in advance.

Edit: I forgot to mention that I'm using .NET 4.6.1 and the latest stable nuget package (2.5.1).

**Comment by haquex19** [comment]
I can confirm this error exists on the latest update on IdentityModel to 3.4.1. When using IdentityModel version 3.3.1, the exception is not raised, but when using package IdentityModel 3.4.1, the exception is raised.

**Comment by leastprivilege** [comment]
Looking into it..

**Comment by leastprivilege** [comment]
I just pushed a 2.6.0 preview to nuget and tried it with WinForms - works fine for me.

**Comment by b-esf** [comment]
I haven't had a chance to try out the 2.6.0 package but after updating some of the associated projects in my solution to .NET 4.7, it seems to have done the trick. I'll give it a go with the 2.6.0 package when I have some time, but we can probably close this one for now.

Thanks.

**Comment by alanisaac** [comment]
I was seeing the exact same issue on my project that is currently targeting .NET Framework 4.6.1.  Based on the comments above, I tried downgrading versions of IdentityModel, but that didn't work for me.  What did work was adding a binding redirect for System.Net.Http.  Unlike the redirect mentioned above, I used 4.2.0.0 as the max for the oldVersion based on [the advice here](http://gertjanvanmontfoort.blogspot.com/2017/11/systemnethttp-dll-version-problems.html):

```
      <dependentAssembly>
        <assemblyIdentity name="System.Net.Http" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-4.2.0.0" newVersion="4.1.1.2" />
      </dependentAssembly>
```
